src/main/java/net/erv123/shadertoymc/arucas/extension/ShaderExtension.java

```diff
@@ -0,0 +1,138 @@
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+@ExtensionDoc(
+	name = "ShaderExtension",
+	desc = "Extension with some functions for interacting with the Minecraft world.",
+	language = Util.Language.Java
+)
+public class ShaderExtension implements ArucasExtension {
+	@NotNull
+	@Override
+	public List<BuiltInFunction> getBuiltInFunctions() {
+		return List.of(
+			BuiltInFunction.of("getWorld", this::getWorld),
+			BuiltInFunction.of("place", 4, this::placeDefault),
+			BuiltInFunction.of("place", 5, this::placeWithWorld),
+			BuiltInFunction.of("area", 7, this::area)
+		);
+	}
+
+	@NotNull
+	@Override
+	public String getName() {
+		return "ShaderExtension";
+	}
+
+	private String getWorld(Arguments arguments) {
+		return ScriptUtils.getScriptHolder(arguments.getInterpreter()).getWorld().getRegistryKey().getValue().getPath();
+	}
+
+	private Void placeDefault(Arguments arguments) {
+		String block = arguments.nextPrimitive(StringDef.class);
+		int x = arguments.nextPrimitive(NumberDef.class).intValue();
+		int y = arguments.nextPrimitive(NumberDef.class).intValue();
+		int z = arguments.nextPrimitive(NumberDef.class).intValue();
+		ServerCommandSource source = ScriptUtils.getScriptHolder(arguments.getInterpreter());
+		this.place(arguments.getInterpreter(), source.getWorld(), block, x, y, z);
+		return null;
+	}
+
+	private Void placeWithWorld(Arguments arguments) {
+		String block = arguments.nextPrimitive(StringDef.class);
+		int x = arguments.nextPrimitive(NumberDef.class).intValue();
+		int y = arguments.nextPrimitive(NumberDef.class).intValue();
+		int z = arguments.nextPrimitive(NumberDef.class).intValue();
+		String worldString = arguments.nextPrimitive(StringDef.class);
+
+		RegistryKey<World> registry = RuntimeError.wrap(() -> RegistryKey.of(RegistryKeys.WORLD, Identifier.tryParse(worldString)));
+		ServerCommandSource source = ScriptUtils.getScriptHolder(arguments.getInterpreter());
+		ServerWorld world = source.getServer().getWorld(registry);
+
+		if (world == null) {
+			throw new RuntimeError("Failed to get world for: " + worldString);
+		}
+		this.place(arguments.getInterpreter(), world, block, x, y, z);
+		return null;
+	}
+
+	private Void area(Arguments arguments) {
+		int ax = arguments.nextPrimitive(NumberDef.class).intValue();
+		int ay = arguments.nextPrimitive(NumberDef.class).intValue();
+		int az = arguments.nextPrimitive(NumberDef.class).intValue();
+		int bx = arguments.nextPrimitive(NumberDef.class).intValue();
+		int by = arguments.nextPrimitive(NumberDef.class).intValue();
+		int bz = arguments.nextPrimitive(NumberDef.class).intValue();
+		ClassInstance callback = arguments.nextFunction();
+		Interpreter interpreter = arguments.getInterpreter();
+		int startX = Math.min(ax, bx);
```

Comment:
input needs to be origin and size not pos1 and pos2

Reply:
Sure.